File: web/src/app/model/attribute-color.ts
export class AttributeColor {

    id: number;

    color: string;

    condition: string;

    static from(value: any): AttributeColor {
        let attributeColor = new AttributeColor();
        attributeColor.id = value.id;
        attributeColor.color = value.color;
        attributeColor.condition = value.condition;
        return attributeColor;
    }

    toSave(): any {
        return {
            id: this.id,
            color: this.color,
            condition: this.condition
        }
    }

}

File: server/src/main/java/com/evgenltd/lt/entity/AttributeColor.java
package com.evgenltd.lt.entity;

import javax.persistence.*;

@Entity
@Table(name = "attribute_colors")
public class AttributeColor {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;

    @ManyToOne
    @JoinColumn(name = "attribute_id")
    private Attribute attribute;

    private String color;

    private String condition;

    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    public Attribute getAttribute() {
        return attribute;
    }

    public void setAttribute(final Attribute attribute) {
        this.attribute = attribute;
    }

    public String getColor() {
        return color;
    }

    public void setColor(final String color) {
        this.color = color;
    }

    public String getCondition() {
        return condition;
    }

    public void setCondition(final String condition) {
        this.condition = condition;
    }
}
